doc/maximal.xml

```diff
@@ -10,102 +11,194 @@
-  <subsemigroup of 6x4 Rees 0-matrix semigroup with 21 generators>, 
-  <subsemigroup of 6x4 Rees 0-matrix semigroup with 23 generators>, 
-  <subsemigroup of 6x4 Rees 0-matrix semigroup with 21 generators>, 
-  <subsemigroup of 6x4 Rees 0-matrix semigroup with 21 generators>, 
-  <subsemigroup of 6x4 Rees 0-matrix semigroup with 21 generators> ]]]></Log>
+    The behaviour of the operation <C>MaximalSubsemigroups</C> can be altered
+    via the second argument <A>opts</A>, which should be a record.
+
+    <List>
+
+      <Mark>number</Mark>
+      <Item>
+        If <C><A>opts</A>.number</C> is <K>true</K>, then
+        <C>MaximalSubsemigroups</C> will return the number of maximal
+        subsemigroups that it finds, without having to create the subsemigroups
+        themselves. It can be significantly faster to compute the number of
+        maximal subsemigroups of a semigroup than to create the maximal
+        subsemigroups as well.  Using <C>MaximalSubsemigroups</C> with the
+        option <C><A>opts</A>.number</C> set to be <K>true</K>, whilst
+        specifying additional options, allows for the extension of the
+        functionality provided by <Ref Attr="NumberMaximalSubsemigroups"/>.
+
+        <P/>
+
+        If <C><A>opts</A>.number</C> is <K>false</K>, then (as per
+        <C><A>opts</A>.gens</C>) <C>MaximalSubsemigroups</C> will return either
+        the maximal subsemigroups of <A>S</A> themselves, or a generating set
+        for each maximal subsemigroup of <A>S</A>.
+
+        <P/>
+
+        The default value for this option is <K>false</K>.
+      </Item>
+
+      <Mark>gens</Mark>
+      <Item>
+        <E>This option is only relevant if <C><A>opts</A>.number</C> is
+          <K>false</K> (or unspecified).</E>
+
+        <P/>
+
+        If <C><A>opts</A>.gens</C> is <K>true</K>, then
+        <C>MaximalSubsemigroups</C> will return a generating set for each
+        maximal subsemigroup that it finds.  If <C><A>opts</A>.gens</C> is
+        <K>false</K>, then the maximal subsemigroups will instead be returned as
+        semigroup objects.
+
+        <P/>
+
+        For certain types of semigroup and maximal subsemigroup, it can be more
+        computationally expensive to find a generating set for the maximal
+        subsemigroup than to create the maximal subsemigroup as a semigroup
+        object without knowing a generating set. The default value for this
+        option is <K>false</K>.
+      </Item>
+
+      <Mark>contain</Mark>
+      <Item>
+        If <C><A>opts</A>.contain</C> is a subset of <A>S</A>, then
+        <C>MaximalSubsemigroups</C> will search for those maximal subsemigroups
+        of <A>S</A> which contain the set <C><A>opts</A>.contain</C>.
+      </Item>
+
+      <Mark>D</Mark>
+      <Item>
+        For a maximal subsemigroup <M>M</M> of a finite semigroup <A>S</A>,
+        there exists a unique &D;-class which contains <M><A>S</A> \setminus
```

Comment:
Try to avoid using the `<M>` tags as they don't display well in the text version of the manual. I think it is better to write out what you mean, in this example, `which contains the complement of <C>M</C> in <A>S</A>.`